### ISSUE TITLE
Add XSLT Transform Ability

### ISSUE THREAD

**Issue by JasonKoopmans** [DotJoshJohnson/vscode-xml#159]
Is there any interest in adding a feature that would allow execution of XSLT transforms? I don't have experience in developing vscode plugins, but I'm willing to contribute if there is an interest in adding the feature.

**Comment by DotJoshJohnson** [comment]
@JasonKoopmans - Sure! I will be merging PR #141 and releasing `v2.0.0` here in the next week. Once that is done, that'd be a good time to pull in something like this (I changed the structure of the repo quite a bit in that branch). I'll follow up then and we can determine the best way to tackle this.

Thanks!

**Comment by JasonKoopmans** [comment]
Cool, I'll look that PR over and get acquainted.

**Comment by kdcarlisle42** [comment]
You should look at two options.  A native xslt transformation and a call out to an external application (similar to xquery).  A lot of times we do basic xslt transforms so a quick native way is would be great, but we also have a small command line tool that will perform a transformation.  Our tool includes additional xslt functions that are made available to the xsl, so when we using those functions, we have to use the tool, rather than a native transform.

**Comment by JasonKoopmans** [comment]
@DotJoshJohnson so do you see this getting implemented like the xquery-execution feature?  Other suggestions/advice?

**Comment by mika76** [comment]
@JasonKoopmans  I see you also requested this, same as #93 - I would personally love this extension.

Could this be useful: https://github.com/fiduswriter/xslt-processor  ? or something like https://sarissa.sourceforge.io/ ?

I would suggest a preview type feature if possible - you could use a linked xslt file or prompt the user for one.

Would probably be useful to run on multiple files also, so you could build documentation or other such items.

**Comment by DotJoshJohnson** [comment]
I'm closing this in favor of issue #93. There is an open PR (#247, from @szul) I'll be reviewing to include in the v2.5 release, which will begin implementing these features.
